internal/cli/pipeline_inspect.go

```diff
@@ -0,0 +1,152 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/posener/complete"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type PipelineInspectCommand struct {
+	*baseCommand
+
+	flagJson       bool
+	flagPipelineId string
+}
+
+func (c *PipelineInspectCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+	); err != nil {
+		return 1
+	}
+
+	var pipelineId string
+	if len(c.args) == 0 {
+		c.ui.Output("Pipeline ID required.\n\n%s", c.Help(), terminal.WithErrorStyle())
+		return 1
+	} else {
+		pipelineId = c.args[0]
+	}
+
+	// Pre-calculate our project ref
+	projectRef := &pb.Ref_Project{Project: c.flagProject}
+	if c.flagProject == "" {
+		if c.project != nil {
+			projectRef = c.project.Ref()
+		}
+
+		if projectRef == nil {
+			c.ui.Output("You must specify a project with -project or be inside an existing project directory.\n"+c.Help(),
+				terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	resp, err := c.project.Client().GetPipeline(c.Ctx, &pb.GetPipelineRequest{
+		Pipeline: &pb.Ref_Pipeline{
+			Ref: &pb.Ref_Pipeline_Id{
+				Id: &pb.Ref_PipelineId{
+					Id: pipelineId,
+				},
+			},
+		},
+	})
+	if err != nil {
+		if status.Code(err) == codes.NotFound {
+			c.ui.Output("Pipeline not found: %s", clierrors.Humanize(err),
+				terminal.WithErrorStyle())
+			return 1
+		}
+
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+	if resp == nil {
+		c.ui.Output("The requested pipeline response was empty", terminal.WithWarningStyle())
+		return 0
+	}
+
+	if c.flagJson {
+		var m jsonpb.Marshaler
+		m.Indent = "\t"
+		str, err := m.MarshalToString(resp)
+		if err != nil {
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+
+		fmt.Println(str)
+		return 0
+	}
+
+	var owner string
```

Comment:
Yeah, so in the context of a pipeline, a Pipeline owner is the project and pipeline name. In the future, it could be something else (we could make it a OneOf), but for now owner is which project a pipeline belongs to and its unique name.